Firmware/motion_control.cpp

```diff
@@ -56,7 +56,7 @@ void mc_arc(const float* position, float* target, const float* offset, float fee
     if (cs.arc_segments_per_sec > 0)
     {
         // 20200417 - FormerLurker - Implement MIN_ARC_SEGMENTS if it is defined - from Marlin 2.0 implementation
-        float mm_per_arc_segment_sec = (feed_rate / 60.0f) * (1.0f / cs.arc_segments_per_sec);
+        float mm_per_arc_segment_sec = feed_rate / (60.f * float(cs.arc_segments_per_sec));
```

Comment:
this change also looks correct to me:
- `feed_rate` is a `float`
- `cs.arc_segments_per_sec` will be upcast to `float`, may be our soft FPU implementation already has a dedicated routine for multiplying `float` * `int`.
- the rest is just reordering of fraction `float` numbers

Reply:
8 bytes shorter:
```
float mm_per_arc_segment_sec = feed_rate * ((1.f/60.f) * cs.arc_segments_per_sec);
		ldi r25, 0x00 ; 0 
		ldi r24, 0x00 ; 0 
		call <__floatunsisf>
		ldi r18, 0x89 ; 137 
		ldi r19, 0x88 ; 136 
		ldi r20, 0x88 ; 136 
		ldi r21, 0x3C ; 60 
		call <__mulsf3>
		adiw r28, 0x2a ; 42 
		ldd r18, Y+60 ; 0x3c 
		ldd r19, Y+61 ; 0x3d 
		ldd r20, Y+62 ; 0x3e 
		ldd r21, Y+63 ; 0x3f 
		sbiw r28, 0x2a ; 42 
		call <__mulsf3>
```

Reply:
Here you're omitting the division `1 / cs.arc_segments_per_sec` which is probably saving those 8 bytes

Reply:
:thinking: may be you are right, this is getting too :exploding_head: